operators/src/source/gdal_source.rs

```diff
@@ -522,8 +354,14 @@ where
         query: crate::engine::QueryRectangle,
         _ctx: &'a dyn crate::engine::QueryContext,
     ) -> BoxStream<Result<RasterTile2D<T>>> {
-        self.tile_stream(query.bbox, query.time_interval, query.spatial_resolution)
-            .boxed() // TODO: handle query, ctx, remove one boxed
+        let meta_data = self.meta_data.loading_info(query).unwrap(); // TODO: change when query signature returns Result
+
+        // TODO: more efficient chaining of streams
```

Comment:
what is not efficient here?

Reply:
I wasn't sure whether the streams are created lazily. If they are I can delete the comment.

Reply:
how can we figure this out?

Reply:
I traced it:
> running 1 test
create tile stream
load tile
load tile
load tile
load tile
create tile stream
load tile
load tile
load tile
load tile
test source::gdal_source::tests::test_query_multi_time_slices ... ok

so it seems fine to me. I removed the todos 51d34bd866d428217421543782d1298306833141